### PR TITLE
Api 13057 add icn parameter to practitioner search

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-13057

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/practitioner/R4PractitionerController.java [hunk 1]
```diff
@@ -152,19 +156,21 @@ String readRaw(@PathVariable("publicId") String publicId, HttpServletResponse re
   }
 
   private boolean tokenIdentifierIsSupported(TokenParameter token) {
-    return token.hasAnySystem()
-        || (token.hasSupportedSystem(PRACTITIONER_IDENTIFIER_SYSTEM_NPI)
-            && token.hasExplicitCode());
+    return token.hasSupportedSystem(
+            PRACTITIONER_IDENTIFIER_SYSTEM_ICN, PRACTITIONER_IDENTIFIER_SYSTEM_NPI)
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/practitioner/R4PractitionerController.java:160]: You need to keep `&& token.hasExplicitCode()`

File: data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/PractitionerIT.java [hunk 2]
```diff
@@ -61,12 +67,36 @@ void searchByIdentifier() {
...
+            200,
+            Practitioner.Bundle.class,
+            b -> isEmpty(b),
+            "Practitioner?identifier=http://va.gov/mpi|{icn}",
             testIds.unknown()),
         test(
             200,
             Practitioner.Bundle.class,
             b -> isEmpty(b),
-            "Practitioner?identifier=http://hl7.org/fhir/sid/us-npi|"),
```
Comment (agarcia-va) [inline data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/PractitionerIT.java:69]: Why was this test removed? Perhaps it should be put in a `searchByIdentifierExplicitSystemNoCode()` test.

**Review by agarcia-va — APPROVED** [review]
LGTM :shipit:

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitioner/R4PractitionerControllerTest.java [hunk 3]
```diff
@@ -51,6 +51,7 @@ R4PractitionerController _controller() {
   @SuppressWarnings("unchecked")
   @ValueSource(
       strings = {
+        "?identifier=http://va.gov/mpi|",
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitioner/R4PractitionerControllerTest.java:54]: There should also be a corresponding `validRequest()` test case

File: data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/PractitionerIT.java [hunk 4]
```diff
@@ -90,7 +120,38 @@ void searchByIdentifier() {
             Practitioner.Bundle.class,
             b -> isEmpty(b),
             "Practitioner?identifier=unknown-system|{npi}",
-            testIds.practitioners().npi()),
+            testIds.practitioners().npi()));
+  }
+
+  @Test
+  void searchByIdentifierExplicitSystemNoCode() {
```
Comment (jblefkove-va) [inline data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/PractitionerIT.java:127]: `ExplicitSystemAnyCode`

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**